testsuite/integration-arquillian/servers/auth-server/services/testsuite-providers/src/main/java/org/keycloak/testsuite/federation/UserPropertyFileStorage.java

```diff
@@ -146,88 +144,64 @@ public int getUsersCount(RealmModel realm) {
     }
 
     @Override
-    public List<UserModel> getUsers(RealmModel realm) {
-        List<UserModel> users = new LinkedList<>();
-        for (Object username : userPasswords.keySet()) {
-            users.add(createUser(realm, (String)username));
-        }
-        return users;
+    public Stream<UserModel> getUsersStream(RealmModel realm) {
+        return userPasswords.keySet().stream().map(obj -> createUser(realm, (String) obj));
     }
 
     @Override
-    public List<UserModel> searchForUser(Map<String, String> attributes, RealmModel realm) {
-        return searchForUser(attributes, realm, 0, Integer.MAX_VALUE - 1);
+    public Stream<UserModel> searchForUserStream(Map<String, String> attributes, RealmModel realm) {
+        return searchForUserStream(attributes, realm, 0, Integer.MAX_VALUE - 1);
     }
 
     @Override
-    public List<UserModel> getUsers(RealmModel realm, int firstResult, int maxResults) {
-        if (maxResults == 0) return Collections.EMPTY_LIST;
-        List<UserModel> users = new LinkedList<>();
-        int count = 0;
-        for (Object un : userPasswords.keySet()) {
-            if (count++ < firstResult) continue;
-            String username = (String)un;
-            users.add(createUser(realm, username));
-            if (users.size() + 1 > maxResults) break;
-        }
-        return users;
+    public Stream<UserModel> getUsersStream(RealmModel realm, int firstResult, int maxResults) {
+        return userPasswords.keySet().stream().skip(firstResult < 0 ? 0 : firstResult).limit(maxResults < 0 ? 0 : maxResults)
```

Comment:
What "-1" in `maxResults` actually should achieve? If that means we don't want limit number of results shouldn't we use `Integer.MAX_VALUE - 1` ? Not sure if `limit(0)` will do the same. 
If "-1" means I don't want any results (which doesn't make sense to me :D ) then it's, I guess correct.

Reply:
I will have to check this again, this is a very good point.